### ISSUE TITLE
How does Dissonance actually sends data over Photon Pun? 

### ISSUE THREAD

**Issue by friuns2** [Placeholder-Software/Dissonance#237]
Hi again ;)

I started debugging how PhotonCommsNetwork sends data

When client speaks does it sends data first to MasterClient, and then master client sends data to other clients? This means data goes twice through server doubling traffic.

Is there way to switch mode to peer to peer to avoid this?  

Dissonance 7.1.0

**Comment by martindevans** [comment]
Dissonance networking has a couple of modes of operation:
 - The default mode is for clients to send all data relayed via the server. In this mode the clients still calculate everything (who to send to etc), the server is simply acting as a dumb relay.
 - In some backends, Clients attempt to establish direct communication with other clients. This is allowed to fail, in which case data is still relayed via the server.

In PUN, the "p2p" connections should be established for every client to every other client and it should never fail (because it's still sent via the PUN network system, which guarantees it'll work). Look for `PhotonClient.OnServerAssignedSessionId` being called, which is where clients send a greeting to other clients. You can see P2P packets being sent in `PhotonClient.SendP2P` which sends directly too clients which is can, and then relays the rest via the server.

tl;dr: Voice packets are only sent through the server during initial session setup.

**Comment by friuns2** [comment]
I see p2p packet 100bytes is send to server and to others, on PUN it means it sends same packet twice to server
![image](https://user-images.githubusercontent.com/16543239/130351203-bf8ea5ec-5820-43df-9e34-fc4f0c5bdb1d.png)

here p2p packets relayed through server
![image](https://user-images.githubusercontent.com/16543239/130351296-c0df9f08-a622-48e2-bb1d-2cafb43466f4.png)

is it possible disable this relay functionality?

**Comment by friuns2** [comment]
Also i think you have typo in Pun1 integration (not sure if you support it anymore, i couldn't find it in Dissonance 8.1)  
 

![](https://user-images.githubusercontent.com/16543239/130352846-908b0ded-3a16-4f73-a387-4cdde08db7e5.png)

I commented out those base implementation and everything still works, with no relay sended. not sure if it breaks anything latter :)

**Comment by martindevans** [comment]
Just to double check before I debug this any more - are you using PUN1 or PUN2? We deprecated the PUN1 integration a while ago.

**Comment by martindevans** [comment]
It's been a while so I'll close this discussion, feel free to re-open it if you have more questions :)
